From 453825ed1b3bc13ab41d8a381056852f20ed1c32 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:03 +0800
Subject: [PATCH 036/150] dt-bindings: hwspinlock: Add binding documentation
 for Phytium hwspinlock

This patch adds the binding documentation for Phytium hardware semaphore
device.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I2aa6693f5cc9153efd8c72f54f628344b9f94906
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/hwlock/phytium,hwspinlock.yaml   | 50 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 51 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml

diff --git a/Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml b/Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml
new file mode 100644
index 000000000000..e361a791627f
--- /dev/null
+++ b/Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/hwlock/phytium,hwspinlock.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium HwSpinlock Driver
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,hwspinlock
+
+  reg:
+    maxItems: 1
+    description: Contains the hwspinlock module register address space.
+
+  "#hwlock-cells":
+    const: 1
+    description: |
+      The Phytium hwspinlock users will use a 0-indexed relative hwlock number as
+      the argument specifier value for requesting a specific hwspinlock within
+      a hwspinlock bank.
+
+      Please look at the generic hwlock binding for usage information for
+      consumers, "Documentation/devicetree/bindings/hwlock/hwlock.txt"
+
+  nr-locks:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    description: The number of locks in the device.
+
+required:
+  - compatible
+  - reg
+  - "#hwlock-cells"
+  - nr-locks
+
+additionalProperties: false
+
+examples:
+
+  - |
+    hwspinlock: spinlock@40000000 {
+      compatible = "phytium,hwspinlock";
+      reg = <0x40000000 0x1000>;
+      #hwlock-cells = <1>;
+      nr-locks = <32>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index cd8b0e188abb..a94ab94c5df8 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2531,6 +2531,7 @@ M:	Chen Baozi <chenbaozi@phytium.com.cn>
 S:	Maintained
 W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
+F:	Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml
 F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
 F:	Documentation/devicetree/bindings/iio/adc/phytium,adc.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,mci.yaml
-- 
2.47.0

